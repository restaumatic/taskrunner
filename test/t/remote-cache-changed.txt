# no toplevel
# s3

export TASKRUNNER_SAVE_REMOTE_CACHE=1

mkdir a

(
  cd a
  echo foo > input.txt
  git init -q
  git add input.txt
)

cp -r a b

go() {
  # Note: we want separate workdir for separate repos
  TASKRUNNER_STATE_DIRECTORY="$(pwd)" taskrunner -n mytask bash -e -c '
    snapshot input.txt --outputs output.txt
    echo "Expensive computation"
    rev input.txt > output.txt
  '
}

(
  cd a
  go
  echo "Output in a is: $(cat output.txt)"
)

(
  cd b
  echo bar > input.txt
  go # note: should run "expensive computation" again
  echo "Output in b is: $(cat output.txt)"
)
