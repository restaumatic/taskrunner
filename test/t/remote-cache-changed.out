-- output:
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation
[mytask] info | success
Output in a is: oof
[mytask] info | Inputs changed, running task
[mytask] stdout | Expensive computation
[mytask] info | success
Output in b is: rab
